src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/LanguageServices/EditorConfigFileWatcher.cs

```diff
@@ -0,0 +1,271 @@
+
+using Task = System.Threading.Tasks.Task;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.LanguageServices
+{
+    /// <summary>
+    ///     Watches for adds and deletes of .editorconfig files *above* the project
+    ///     directory and triggers evaluation as needed. Note that adds and deletes of
+    ///     files within the project directory directory already trigger evaluation, so
+    ///     this type doesn't handle that case.
+    /// </summary>
+    internal class EditorConfigFileWatcher : OnceInitializedOnceDisposedAsync, IVsFreeThreadedFileChangeEvents2
+    {
+        private static readonly TimeSpan s_notifyDelay = TimeSpan.FromMilliseconds(100);
+
+        private readonly IActiveConfiguredProjectSubscriptionService _activeConfiguredProjectSubscriptionService;
+        private readonly IVsService<IVsAsyncFileChangeEx> _fileChangeService;
+        private readonly IProjectTreeProvider _fileSystemTreeProvider;
+        private readonly IUnconfiguredProjectCommonServices _projectServices;
+        private readonly IUnconfiguredProjectTasksService _projectTasksService;
+
+        private Dictionary<string, uint> _filesBeingWatched;
+        private IDisposable _treeWatcher;
+        private ITaskDelayScheduler _taskDelayScheduler;
+        private CancellationTokenSource _taskDelaySchedulerCancellationTokenSource;
+
+        [ImportingConstructor]
+        public EditorConfigFileWatcher(
+            IVsService<SVsFileChangeEx, IVsAsyncFileChangeEx> fileChangeService,
+            [Import(ContractNames.ProjectTreeProviders.FileSystemDirectoryTree)] IProjectTreeProvider fileSystemTreeProvider,
+            IUnconfiguredProjectCommonServices projectServices,
+            IUnconfiguredProjectTasksService projectTasksService,
+            IActiveConfiguredProjectSubscriptionService activeConfiguredProjectSubscriptionService)
+            : base(projectServices.ThreadingService.JoinableTaskContext)
+        {
+            Requires.NotNull(fileChangeService, nameof(fileChangeService));
+            Requires.NotNull(fileSystemTreeProvider, nameof(fileSystemTreeProvider));
+            Requires.NotNull(projectServices, nameof(projectServices));
+            Requires.NotNull(projectTasksService, nameof(projectTasksService));
+            Requires.NotNull(activeConfiguredProjectSubscriptionService, nameof(activeConfiguredProjectSubscriptionService));
+
+            _fileChangeService = fileChangeService;
+            _fileSystemTreeProvider = fileSystemTreeProvider;
+            _projectServices = projectServices;
+            _projectTasksService = projectTasksService;
+            _activeConfiguredProjectSubscriptionService = activeConfiguredProjectSubscriptionService;
+        }
+
+#pragma warning disable RS0030 // Do not used banned APIs
+        [ConfiguredProjectAutoLoad(RequiresUIThread = false)]
+#pragma warning restore RS0030 // Do not used banned APIs
+        [AppliesTo(ProjectCapability.DotNet)]
+        internal void Load()
```

Comment:
This is loading for Shared Projects, which is probably not want you want. You want DotNetLanguageService for those projects just with the LanguageService.